### ISSUE TITLE
ice.so module missing from 20.04 php-zeroc-ice package

### ISSUE THREAD

**Issue by cjgordon** [zeroc-ice/ice#1338]
**Version:**
 - Ice Version 3.7.7-1
 - OS: Ubuntu 20.04
 - Distribution: Apt

**Describe the bug**
When installing php-zeroc-ice on platform noted above it does not included ice.so extension for PHP8. It is indicated on this page: https://doc.zeroc.com/ice/latest/release-notes/supported-platforms-for-ice-3-7-7 that PHP 8.0 should be supported.

**Expected behavior**
After installing package php-zeroc-ice and enabling ice extension that PHP info would show Ice module as being loaded:
```
    $ apt-get -s install php-zeroc-ice
    Reading package lists... Done
    Building dependency tree       
    Reading state information... Done
    php-zeroc-ice is already the newest version (3.7.7-1).

    $ php -i 
    PHP Warning:  PHP Startup: Unable to load dynamic library 'ice.so' (tried: /usr/lib/php/20200930/ice.so (/usr/lib/php/20200930  /ice.so: cannot open shared object file: No such file or directory), /usr/lib/php/20200930/ice.so.so (/usr/lib/php/20200930/ice.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
```

**Additional context**
When investigating this further the folder /usr/lib/php/20190902 contains ice.so which is the version built for php7.4.

The folders /usr/lib/php/20200930 and /usr/lib/php/20210902 do not contain ice.so. These folders contain versions of extensions built for 8.0 and 8.1 respectively.

You cannot use the version in the folder 20190902 with 8.0 as it has been built using wrong version.

Pretty sure I can build the extension myself for PHP8 as I have done this for previous PHP versions of ice when this same issue has occurred.

EDIT: More appropriate title.

**Comment by pepone** [comment]
Hi,

The apt and rpm packages only include the PHP extension for the default PHP distribution. The supported platforms page mention 8.0 and 8.1 because we have test source builds for this PHP versions.

We might consider including several versions for a future release if this is common practice, do you know of other packages that include binary extension for several PHP versions?

Cheers,
Jose

**Comment by cjgordon** [comment]
Hi @pepone,

Thanks for the response.

I was able to build it for PHP8.0 and 8.1 from source as expected. I guess the current behavior is okay but it didn't seem right when following the installation documentation here:
[https://zeroc.com/downloads/ice/3.7/php#linux](https://zeroc.com/downloads/ice/3.7/php#linux)

Maybe there just needs to be a note somewhere that it will only work for platform supported version of PHP and you  need to build it manually if not using platform supported version.

I had installed php8.0 using: https://launchpad.net/~ondrej/+archive/ubuntu/php which I think it pretty common practice. For most other extensions when you do this you install the appropriate version using the version in the package name like: php8.0-sodium for example.

Happy to close this. 

Cheers,
Chris
